dashboard/app/models/user.rb

```diff
@@ -1055,12 +1055,12 @@ def next_unpassed_visible_progression_level(script)
     # will be redirected to /congrats.
     return nil if completed?(script)
 
-    visible_sls = visible_script_levels(script).reject {|sl| sl.bonus || sl.level.unplugged? || sl.locked?(self)}
+    visible_sls = visible_script_levels(script).reject {|sl| sl.bonus || sl.level.unplugged?}
 
     sl_level_ids = visible_sls.map(&:level_ids).flatten
 
     # Levels the user made progress in
-    ul_level_ids = user_levels_by_level(script).keys
+    ul_level_ids = user_levels.where(script: script).map(&:level_id)
```

Comment:
Nit: Prefer `pluck` in cases like this:
```suggestion
    ul_level_ids = user_levels.where(script: script).pluck(:level_id)
```
Because it will directly return the results of a `select level_id from...` query, without instantiating ActiveRecord model objects for all of the retrieved UserLevels.